### PR TITLE
CMake: Do not configure CGAL with bundled boost

### PR DESCRIPTION
**Pull request by tamiko**
Fixes: https://cdash.dealii.43-1.org/build/1730/configure

### CODE REVIEW

**Comment by drwells** [comment]
Shouldn't this be fixed by renaming the file to `configure_3_cgal.cmake`?

File: cmake/configure/configure_cgal.cmake [hunk 1]
```diff
@@ -17,4 +17,41 @@
...
 #
 
+IF(NOT FEATURE_BOOST_PROCESSED)
+  MESSAGE(FATAL_ERROR "\n"
+    "Internal build system error: The configuration of "
+    "DEAL_II_WITH_CGAL depends on "
+    "DEAL_II_WITH_BOOST, but CONFIGURE_FEATURE(CGAL) "
+    "was called before CONFIGURE_FEATURE(BOOST).\n\n"
+    )
+ENDIF()
```
Comment (tamiko) [inline cmake/configure/configure_cgal.cmake:27]: @drwells This is just a safeguard ensuring that the order is maintained after configure file renaming :smile_cat:

**Review by drwells — APPROVED** [review]

File: cmake/configure/configure_cgal.cmake (hunk 1 above)
Comment (drwells) [reply cmake/configure/configure_cgal.cmake:27]: Yes, we should definitely still do this!

**Review by luca-heltai — APPROVED** [review]

**Pull request merged**